source/opt/strip_debug_info_pass.cpp

```diff
@@ -22,10 +22,19 @@ Pass::Status StripDebugInfoPass::Process() {
   bool modified = !context()->debugs1().empty() ||
                   !context()->debugs2().empty() ||
                   !context()->debugs3().empty();
+  for (auto& dbg : context()->debugs1()) context()->KillNamesAndDecorates(&dbg);
```

Comment:
I think it would be best to just call `KillInst`.  I know this pass does not keep any of the analyses up-to-date, but it is still the standard way of deleting an instruction.

Reply:
Done